charts/4allportal/templates/samba/deployment.yaml

```diff
@@ -0,0 +1,142 @@
+spec:
+  strategy:
+    type: Recreate
+  replicas: 1
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" $ | nindent 6 }}
+      app.kubernetes.io/component: samba
+  template:
+    metadata:
+      labels: {{- include "common.labels.stable" $ | nindent 8 }}
+        app.kubernetes.io/component: samba
+    spec:
+      {{- include "4allportal.samba.priorityClassName" $ | nindent 6 }}
+      containers:
+        - name: samba
+          image: {{ template "4allportal.samba.image" $ }}
+          {{- if $.Values.samba.image.repository | contains "@" }}
+          imagePullPolicy: IfNotPresent
+          {{- else }}
+          imagePullPolicy: Always
+          {{- end }}
+          ports:
+            - containerPort: 445
+              name: smb
+          command: [ "sh", "-c" ]
+          args:
+            - |
+              useradd -u 1000 4allportal
+              samba-container run smbd
+          volumeMounts:
+            - mountPath: /etc/container-config
+              name: samba-container-config
+            - mountPath: /etc/container-users
+              name: users-config
+            - mountPath: /mnt/assets
+              name: assets
+            - mountPath: /mnt/config
+              name: config
+            - mountPath: /var/lib/samba
+              name: samba-state
+            - mountPath: /run
+              name: run
+            - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+              name: kube-api-access-abcde
+              readOnly: true
+          {{- if $.Values.samba.livenessProbe.enabled }}
+          livenessProbe:
+            exec:
+              command:
+                - bash
+                - -c
+                - timeout 1 ls -d /mnt/assets
+                - timeout 1 ls -d /mnt/config
+            failureThreshold: {{ $.Values.samba.livenessProbe.failureThreshold }}
+            initialDelaySeconds: {{ $.Values.samba.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ $.Values.samba.livenessProbe.periodSeconds }}
+            successThreshold: {{ $.Values.samba.livenessProbe.successThreshold }}
+            timeoutSeconds: {{ $.Values.samba.livenessProbe.timeoutSeconds }}
+          {{- end }}
+          {{- if $.Values.samba.readinessProbe.enabled }}
+          readinessProbe:
+            tcpSocket:
+              port: samba
+            failureThreshold: {{ $.Values.samba.readinessProbe.failureThreshold }}
+            initialDelaySeconds: {{ $.Values.samba.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ $.Values.samba.readinessProbe.periodSeconds }}
+            successThreshold: {{ $.Values.samba.readinessProbe.successThreshold }}
+            timeoutSeconds: {{ $.Values.samba.readinessProbe.timeoutSeconds }}
+          {{- end }}
+          resources: {{- toYaml $.Values.samba.resources | nindent 12 }}
+          env:
+            - name: SAMBA_CONTAINER_ID
+              value: assets
```

Comment:
Not super important, but identifying the config for assets and config(data) with the name assets could be a bit confusing later on.